### PR TITLE
GH Actions

### PR DESCRIPTION
**Pull request by ocefpaf**
The first commit adds the GH actions that check the code style. The second one apply the changes.

For those who have pre-commit installed and configure these changes will be applied automatically on every commit.

Do not merge this one before #82. It requites that PR first.

### CODE REVIEW

**Comment by efiring** [comment]
Wow, that's a lot of code churn!  It certainly brings consistency, but I'm a little reluctant to turn all code formatting over to a robot. Maybe I will be convinced, but it hasn't happened yet.
Why is mypy in there?  Isn't that for type annotations? 
Are all these new dependencies and rigid rules really helpful for a tiny project like this?

**Comment by ocefpaf** [comment]
> Wow, that's a lot of code churn!

Indeed. And we should not merge if it unless everybody agrees on it. Ideally, having everything in a single massive commit helps with `git blame` later. We can just ignore this commit when using it with `--ignore-rev` when using `blame`.

> It certainly brings consistency, but I'm a little reluctant to turn all code formatting over to a robot. Maybe I will be convinced, but it hasn't happened yet.

I was not sold on black at first but after using it in other projects I find that it lifts a weight from my shoulders about thinking, reviewing, and teaching consistency instead of just applying it and worrying about the code. Also, it lowers the bar for new contributors b/c they too don't need to worry about it, the CIs will handle that part either are a review or automatically if you have pre-commit installed. (If you have pre-commit installed the changes will be applied for you when you commit something, so you will always pushed code that passes the style check.)

> Why is mypy in there? Isn't that for type annotations?

Yes but it also help to catches issues even when you don't have type annotations in the code. But we can remove it if in case you want to merge this PR.

> Are all these new dependencies and rigid rules really helpful for a tiny project like this?

That is a good question. The dependencies are development deps only and handled by the CIs. We don't need to worry about them as users. Even as devs we can just "outsource" the checks to the CIs if we don't want to check these things before committing, so no new deps to install at all.

Regarding the size, I find that sometimes having a tidy code increases the changes of getting new contributors and reduce the chances of people just re-writing everything in their own way and fragmenting the community. However, it is definitely not that important here.

TL;DR I'm OK if you choose to close this but I don't believe this brings extra complexity to the project. In fact, it may reduce it when we review future chances.

**Comment by efiring** [comment]
@ocefpaf If we were to hold off on this, say until black is no longer beta and github supports blame while ignoring one or more commits, how hard would it be for you to redo it?  Also, there is one commit in here that is unrelated to the reformatting, correct?
@wesleybowman Do you have any opinions?
I'm still on the fence, but I would like to take some action, either close or merge, so that this doesn't hold up making substantive changes.

**Comment by ocefpaf** [comment]
> @ocefpaf If we were to hold off on this, say until black is no longer beta and github supports blame while ignoring one or more commits, how hard would it be for you to redo it? Also, there is one commit in here that is unrelated to the reformatting, correct?

Quite easy. No need to merge this right now. However, `black` is a special case were the beta version does not mean anything. One does not even need the `--pre` option to install it. Regarding git blame, black devs recommend to just do a single massive commit and then use `git hyper blame` to ignore it.

> @wesleybowman Do you have any opinions?
> I'm still on the fence, but I would like to take some action, either close or merge, so that this doesn't hold up making substantive changes.

BTW, I was lazy and pushed a lot of changes in a single PR. We could probably just do the first commit instead.

**Comment by wesleybowman** [comment]
>  @wesleybowman Do you have any opinions?

For me, `black` is wonderful. We use it at work for our production code. We've now been using it for around a year, and we've had no problems.

**Comment by efiring** [comment]
OK, there we are: merged.  There is one style decision in black that I think is ugly and thoroughly unpythonic: the placement of a trailing parenthesis on a single line, lined up with the first character in the first line of its matching opening parenthesis.
